protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -26,7 +30,36 @@ open class CodeGenerator(
         fn().also { indent = indent.dropLast(4) }
     }
 
-    protected fun writeType(type: File.Type, parentType: String? = null) = when (type) {
+    private fun writeExtension(field: File.Field) {
+        when (field) {
+            is File.Field.Numbered -> {
+                line(
+                    "val ${file.kotlinTypeMappings()[field.extendee!!]}.${field.kotlinFieldName}: ${
+                        field.kotlinValueType(
+                            true
+                        )
+                    }? get() ="
```

Comment:
This is causing the generated code to end up with types like `String??` when the type returned by `field.kotlinValueType()` is already nullable.

I don't think you actually want the type to always be nullable since the `getExtension()` method will return the default value for fields that aren't present. For example, a `repeated` extension field would have a default value of `emptyList()`, so the type of the field should be `List<T>` rather than `List<T>?`. Similarly, a proto3 `int32` would have a default value of `0` and so its type should be `Int` rather than `Int?`, whereas a proto2 `optional int32` would have a default value of `null` and a type of `Int?`. I _think_ this is all handled for you by the `field.kotlinValueType()` method.